pkg/operator/controllers/muo/deploy.go

```diff
@@ -0,0 +1,119 @@
+package muo
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+	"github.com/Azure/ARO-RP/pkg/util/ready"
+)
+
+type deployer struct {
+	kubernetescli kubernetes.Interface
+	dh            dynamichelper.Interface
+}
+
+func newDeployer(kubernetescli kubernetes.Interface, dh dynamichelper.Interface) *deployer {
+	return &deployer{
+		kubernetescli: kubernetescli,
+		dh:            dh,
+	}
+}
+
+func (o *deployer) resources(pullspec string) ([]kruntime.Object, error) {
+	results := []kruntime.Object{}
+	for _, assetName := range AssetNames() {
+		b, err := Asset(assetName)
+		if err != nil {
+			return nil, err
+		}
+
+		obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(b, nil, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// set the image for the deployments
+		if d, ok := obj.(*appsv1.Deployment); ok {
+			if d.Labels == nil {
+				d.Labels = map[string]string{}
+			}
+
```

Comment:
It is a copy paste :D 

I was originally setting the version here since it might have been useful, but ran into a length limitation (because the SHAs are full length coming from Quay, it seems).